speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SDAConfig.cpp

```diff
@@ -98,4 +98,18 @@ void SDAConfig::SetDataCollectionSettings(tDataToStore p_dataSetting)
 tDataToStore SDAConfig::GetDataCollectionSetting() const
 {
     return m_dataCollectionSetting;
+}
+
+/// @brief            Sets the filepath for the black box executable
+/// @param p_filePath A const char* representing the filepath of the black box executable
+void SDAConfig::SetBlackBoxFilePath(const char* p_filePath)
+{
+    strcpy_s(m_blackBoxFilePath, BLACKBOX_PATH_SIZE, p_filePath);
```

Comment:
This is an unsafe copy and will crash the program if the filepath is too long, you need to check whether p_filepath doesnt exceed 256 bytes, either here or when selecting the blackbox.

Reply:
Sanity check has been added (although the COM interface should already provide that internally)